soroban-sdk/src/env.rs

```diff
@@ -469,6 +469,79 @@ use std::{path::Path, rc::Rc};
 #[cfg(any(test, feature = "testutils"))]
 use xdr::{LedgerEntry, LedgerKey, LedgerKeyContractData, SorobanAuthorizationEntry};
 
+#[cfg(any(test, feature = "testutils"))]
+pub struct TestStellarAssetContractUtil {
+    pub address: Address,
+    env: Env,
+    issuer_id: xdr::AccountId,
+}
+
+#[cfg(any(test, feature = "testutils"))]
+impl TestStellarAssetContractUtil {
+    pub fn get_issuer_flags(&self) -> u32 {
+        self.env
+            .host()
+            .with_mut_storage(|storage| {
+                let k = Rc::new(xdr::LedgerKey::Account(xdr::LedgerKeyAccount {
+                    account_id: self.issuer_id.clone(),
+                }));
+
+                let entry = storage.get(
+                    &k,
+                    soroban_env_host::budget::AsBudget::as_budget(self.env.host()),
+                )?;
+
+                match entry.data {
+                    xdr::LedgerEntryData::Account(ref e) => Ok(e.flags.clone()),
+                    _ => panic!("Expected account entry"),
+                }
+            })
+            .unwrap()
+    }
+    /// Sets the issuer flags field.
+    /// Each flag is a bit with values corresponding to [xdr::AccountFlags]
+    ///
+    /// Use this to test interactions between trustlines/balances and the issuer flags.
+    pub fn overwrite_issuer_flags(&self, flags: u32) {
```

Comment:
If this function is just a setter, can we use that term?
```suggestion
    pub fn set_flags(&self, flags: u32) {
```

Reply:
We could, but in the context of these flags, setting is usually a bitwise & operation - https://github.com/stellar/stellar-xdr/blob/4ec28d95dd84b109253e22b151314478d6f00522/Stellar-transaction.x#L215. I didn't want to complicate the api by making it more complex, so just went with a single option to overwrite the value.

Using `set_flags` is fine though. The comment above can explain what it does, and the absence of a `clear_flag` should make it obvious the `set_flags` overwrites the value.

Reply:
I updated the api to now expose `set_flag` and `clear_flag` methods.